lib/runAll.js

```diff
@@ -74,26 +76,15 @@ const runAll = async (
 
   const ctx = getInitialState({ quiet })
 
-  const { gitDir, gitConfigDir } = await resolveGitRepo(cwd)
-  if (!gitDir) {
-    if (!quiet) ctx.output.push(NOT_GIT_REPO)
-    ctx.errors.add(GitRepoError)
-    throw createError(ctx)
-  }
+  // here could be some factory, by default returning git workflow.
+  // repo checks and paths (gitDir, gitConfigDir) belongs to vcsWorkflow entity.
+  const vcsWorkflow = await makeVcsWorkflow({ allowEmpty, matchedFileChunks }, symbols);
 
-  // Test whether we have any commits or not.
-  // Stashing must be disabled with no initial commit.
-  const hasInitialCommit = await execGit(['log', '-1'], { cwd: gitDir })
-    .then(() => true)
-    .catch(() => false)
+  //warnings about backup are also git-specific, can be processed in prepare phase
+  ctx.shouldBackup = await vcsWorkflow.shouldBackup();
 
-  // Lint-staged should create a backup stash only when there's an initial commit
-  ctx.shouldBackup = hasInitialCommit && stash
-  if (!ctx.shouldBackup) {
-    logger.warn(skippingBackup(hasInitialCommit))
-  }
+  const files = await vcsWorkflow.getStagedFilesList({ cwd });
```

Comment:
We could combine files and shouldBackup to an init method after the previous comment:

```js
const { files, shouldBackup } = await workflow.init()
```

Reply:
This should also return a `baseDir` that (in the context of git) resolves points to the repo root (which might be different from the `cwd`.

Reply:
It seems to me that `shouldBackup` is more suitable. Most studios support backups in one way or another. 

> what other features could this init provide? 

It seems that the listed options are enough. In the future, adding is easier than removing. It will be possible to discuss this in a future pull request, when all the details will be more clear to me.